aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/ReadHandler.java

```diff
@@ -23,28 +28,26 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final CallbackContext callbackContext,
             final ProxyClient<SsmClient> proxyClient,
             final Logger logger) {
-        return proxy.initiate("aws-ssm-parameter::resource-read", proxyClient, request.getDesiredResourceState(), callbackContext)
-                .translateToServiceRequest(Translator::getParametersRequest)
-                .makeServiceCall(this::ReadResource)
-                .done((getParametersRequest, getParametersResponse, proxyInvocation, resourceModel, context) -> {
-                    if(getParametersResponse.parameters().size() == 0) {
-                        throw new CfnNotFoundException(ResourceModel.TYPE_NAME, request.getDesiredResourceState().getName());
-                    }
-                    final Parameter parameter = getParametersResponse.parameters().stream().findFirst().get();
-
-                    return ProgressEvent.defaultSuccessHandler(ResourceModel.builder()
-                            .name(parameter.name())
-                            .type(parameter.typeAsString())
-                            .value(parameter.value()).build());
-                });
-    }
+        this.logger = logger;
+        // read model only contains primary key
+        ResourceModel model = request.getDesiredResourceState();
 
-    private GetParametersResponse ReadResource(final GetParametersRequest getParametersRequest,
-                                               final ProxyClient<SsmClient> proxyClient) {
-        try{
-            return proxyClient.injectCredentialsAndInvokeV2(getParametersRequest, proxyClient.client()::getParameters);
-        } catch (final InternalServerErrorException exception) {
+        try {
+            // Get info from getParameters
+            GetParametersResponse getParametersResponse = proxyClient
+                    .injectCredentialsAndInvokeV2(Translator.getParametersRequest(model), proxyClient.client()::getParameters);
+            if (getParametersResponse.parameters().size() == 0) {
+                throw new CfnNotFoundException(ResourceModel.TYPE_NAME, request.getDesiredResourceState().getName());
+            }
+            Parameter parameter = getParametersResponse.parameters().stream().findFirst().get();
```

Comment:
Is this guaranteed to be 1 result always?

Reply:
Hi, yes parameter name is unique and only 1 or 0 result will return